docs/Architecture Decision Record/009-git-workflow.md

```diff
@@ -0,0 +1,94 @@
+    * When changes are asked for, the changes are made by the contributor. Raft reviewers should internalize the changes asked by 18F/HHS such that the same feedback is already incorporated and/or caught in future (continuous improvement)
+    * When security controls are approved, additional reviews are asked for by OCIO
+19. When satisfied, the 18F/HHS/OCIO reviewer `approves` the PR
+20. The contributor clicks Merge into `raft-tdp-main`
+21. The contributor:
+    * opens a PR from `raft-tdp-main` to `HHS:main`
+    * Updates the PR template to change `addresses` to `closes` so that issue [can be automatically closed when 18F/HHS merge](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
+    * Ensures the CI/CD pipelines are green
+    * Assigns 18F/HHS as reviewers and pings them on Teams to review/merge
+26. PR is approved and merged to `HHS:main` by `adpennington` or `lfrohlich` **SLA** 1 day from open date (by now the code is already approved by 18F/HHS)
+
+## Consequences
+
+**Pros**
+- Dependency chains would be much easier to manage
+  - We can daisy chain PRs with dependencies allowing GitHub to manage changes for downline PRs much more simply and cleanly
+  - ie. `my_branch_1` depends on `my_branch_0` and there is an open PR from `my_branch_0` to `raft-tdp-main`. Can open a PR in raft-tech from `my_branch_1` to `my_branch_0` and only the changes from the latest branch will be shown to reviewers, but the dependent code will still be present and kept up to date by GitHub exposing a button to update with upstream branch
+  - Because of this, we can more easily submit smaller PRs since this removes much of the maintenance work currently involved to achieve that goal
+- Less complexity for developers, reviewers
+- No need to rebase from HHS back to `raft-tdp-main`
+- Much smaller chance of needing to revert commits to `HHS:main`
+- Git history will be much cleaner
+- Much less time managing git history
+- All reviews will be in one place, so the entire history of a branch/PR will be viewable in one place
+- Will allow us to keep the current CircleCi setup
+  - Once staging is in place, we can deploy to the development server when code is merged in to `raft-tdp-main` and test deployment ahead of opening a PR to `HHS:main`
+- We can implement it almost immediately
+- We can implement GitHub Hooks to automatically issue PRs to `HHS:main` when one is merged to `raft-tdp-main`
+- Maintain "Least Privelege" by restricting vendor from having write access to Gov repo and CircleCI
+- Tests automated deployment before merging to `HHS:main`
+
+**Cons**
+- Only one pull request at a time will be able to go to `HHS:main`, but this won't be a problem since they will only be issued once approved and therefore can be merged immediately
+- All PR comments will be on the vendor repo rather than the government repo
```

Comment:
appreciate the recommendations presented during the mtg re: carrying over PR comments! think we agreed that developing a more automated solution is something we could revisit down the road. Is this right? Still, it'd be great to have any suggestions you might have on more manual alternatives we could consider and implement in the meantime.

Reply:
Yes, this is something we can definitely investigate. I'm not sure if there is a manual way to move commentary but we can certainly look into it!

Reply:
sounds good. and ill look into this too. one suggestion @alexsoble floated was maybe linking PRs across repos. im not too familiar (yet) with the mechanics of that though. and for what it's worth, this is considered relatively low priority investigations from our end.

Reply:
It would be easy enough to add a link from the PR to HHS, certainly